src/components/Header.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { showError } from '../helpers/helpers';
+import { updateTodo, addTodos, USER_ID } from '../api/todos';
+
+interface Props {
+  todos: Todo[],
+  setError: React.Dispatch<React.SetStateAction<string>>;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setLoader: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setError,
+  setTempTodo,
+  setTodos,
+  setLoader,
+}) => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [submitDisable, setSubmitDisable] = useState(false);
+  const activeInputButton = todos.every(todo => todo.completed);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    (setSearchQuery(event.target.value));
+  };
+
+  const newTodo = {
+    id: 0,
+    title: searchQuery,
+    userId: 10881,
+    completed: false,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setSubmitDisable(true);
+    if (!searchQuery) {
+      showError('Title can\'t be empty', setError);
+      setSubmitDisable(false);
+    } else {
+      setLoader(prevState => [...prevState, newTodo.id]);
+      addTodos(USER_ID, searchQuery)
+        .then(todo => {
+          setTodos(prevState => {
+            setTempTodo(null);
+            setSubmitDisable(false);
+            setSearchQuery('');
```

Comment:
Why are you calling `setTempTodo`, `setSubmitDisable` and `setSearchQuery` in the middle of `setTodos`? 
This actions can be done below in the `.catch` block.

Reply:
When we add todo we create tempTodo. If If the adding was successful, we need to remove tempTodo, remove disable from input and clear input. If we will move this actions to catch block, then they will work only in case of an error